### DISCUSSION TITLE
How to create an Unstructured grid from points and faces

### DISCUSSION THREAD

**Discussion by dthillaithevan** [pyvista/pyvista#2621] [Q&A]
Hi,

I'm trying to generate an Unstructured Grid from a set of points and faces. I've been trying to use the [tutorial](https://docs.pyvista.org/examples/00-load/create-unstructured-surface.html?highlight=unstructuredgrid) and example 
shown [here](https://github.com/pyvista/pyvista-support/issues/442) but haven't been very successful. 

```
import pyvista as pv
import numpy as np
import vtk


# Triangle surface mesh
#Coordinates of nodes,  shape = (n_nodes, 3)
points = np.loadtxt('points.csv',delimiter =',')
#Triangle face definitions, shape=(n_faces, 3)
faces = np.loadtxt('faces.csv',delimiter =',')

n_node = points.shape[0]

elem_ids = np.arange(0, n_node, dtype=np.int32).reshape(-1, 3)

n_elem = elem_ids.shape[0]

padding = np.ones((n_elem, 1), dtype=np.int64)*3
cells = np.hstack((padding, elem_ids)).ravel()

cell_type = np.empty(n_elem, np.uint8)
cell_type[:] = vtk.VTK_TRIANGLE

grid = pv.UnstructuredGrid(cells, cell_type, points)

grid.save('test.vtu')
```

Here's what the structure should look like:
<img width="600" alt="Screenshot 2021-09-30 at 13 01 25" src="https://user-images.githubusercontent.com/44264236/135451664-1a0572c8-9405-444a-be23-df2eb76f2107.png">

Here's what I end up with:
<img width="707" alt="Screenshot 2021-09-30 at 13 01 17" src="https://user-images.githubusercontent.com/44264236/135451689-e2d10e36-2377-4e5f-ae63-360ad50bcdf6.png">

I can't seem to upload a zip file at the moment, but I can share the csv files if needed. 

Thanks in advance!

**Comment by RichardScottOZ** [comment]
Those big triangles etc. suggest something is out of order - maybe upload your sample?

**Comment by akaszynski** [comment]
Element IDs look to be incorrect.
